### ISSUE TITLE
GATE-3 — Phase 3 gate: v1 training parity, black-box suite, and inventory audit

### ISSUE THREAD

**Issue by aacostadiaz** [ACEsuit/mace#1585]
**Depends on:** TRN-1 (#1575), TRN-2 (#1576), TRN-3 (#1577), TRN-4 (#1578), TRN-5 (#1639), CLI-1 (#1579), FT-1 (#1580), FT-4 (#1637), ELEC-1 (#1591), ELEC-2 (#1592), DEP-1 (#1583), DEP-1a (#1634), MAG-1 (#1584) · **Blocks:** —

**Prerequisite:** the Phase 0 characterization suite is committed — goldens, fixtures and the single-source tolerance table under `tests/golden/`, characterization tests under `tests/unit/`. References of the form `P0-N` name that work.

**Context:** Phase 3 delivered the v1 training stack: the config schema (CFG-1 (#1574)), the staged pipeline and explicit training loop (TRN-1 (#1575)), composable losses from observable specs (TRN-2 (#1576)), balancing and metrics (TRN-3 (#1577)), safetensors checkpointing + DDP (TRN-4 (#1578)), the L-BFGS full-batch regime (TRN-5 (#1639)), the hierarchical `mace train/eval/...` CLI (CLI-1 (#1579)), heads-as-modules fine-tuning (FT-1 (#1580)/FT-2 (#1581)/FT-3 (#1582)), the MDP fine-tune mode (FT-4 (#1637)), the electrostatics core — the reference solver and its dispatch layer (ELEC-1 (#1591)) and `PolarMACE` on the new architecture (ELEC-2 (#1592)) — the new ASE calculator and its model-family surfaces (DEP-1 (#1583)/DEP-1a (#1634)), and magnetic MACE (MAG-1 (#1584)). The electrostatics core is in this phase because DEP-1a (#1634), FT-4 (#1637) and this gate consume it: the polar and LES calculator surfaces are DEP-1a (#1634)'s, MDP fine-tuning runs through FT-4 (#1637), and both are Phase 3 deliverables. This ticket is the phase gate: a verification checkpoint, not a feature. It runs the end-to-end evidence that `mace train --engine v1` honors the legacy user contract — the contract pinned in Phase 0 as black-box tests (P0-5) and committed goldens (P0-1, P0-6) — and audits the feature-inventory rows in Phase 2–3 scope. Nothing here writes new production code; failures are fixed in the owning tickets and the gate re-runs. Passing this gate is what allows Phase 4 (foundation-model conversion, JAX) to proceed on a trusted training pipeline; it is also the point from which CI enforces that `git ls-files mace/` (the frozen legacy tree) is strictly non-increasing. The default engine stays `legacy` — the flip to `v1` is a separate, later decision taken only while parity remains green, with legacy retained as the oracle until its retirement tickets delete it.

**Interface & constraints (the seven gate criteria, with their definitions inlined):**

1. **Training parity on the tiny task.** `mace train --engine v1` on the tiny anchor task trains end-to-end: loss decreases, checkpoint + resolved-config metadata are written, resume works, and the final train/valid error table (per-property MAE/RMSE + final loss) is comparable to the committed legacy table `tests/golden/references/tiny_scaleshift_training_errors.json` (produced by the legacy anchor training in P0-1). "Comparable" is deliberately coarse — end-of-training errors are stochastic and cannot see a gradient bug the size of init noise — so the criterion is **reinforced by PAR-1 (#1572)'s single-step, init-agnostic gradient parity test**: byte-identical weights in both stacks, one forward+backward on an energy+forces loss, `d(loss)/d(params)` allclose at fp64 (this exercises the `create_graph=True` second-derivative path). That test must be green for the gate. Beyond the tiny anchors, the legacy-vs-v1 training comparison runs on the dedicated end-to-end regression set assembled in P0-5 (small but diverse: surfaces and molecules, including dipoles and partial charges), old-vs-new with the same metrics.
2. **Both training regimes are covered.** Criterion 1 is a mini-batch Adam run. The gate additionally runs the **two-stage Adam→L-BFGS** schedule (TRN-5 (#1639)) on the tiny task and records that its full-batch gradient matches the concatenated-dataset gradient at fp64 — the regime exists precisely because it is not the mini-batch one, so a gate that only exercises Adam certifies half the pipeline.
3. **Eval and calculator agree.** `mace eval` (CLI-1 (#1579)) and the new ASE calculator (DEP-1 (#1583)) agree numerically per configuration, on the P0-5 contract keys (`energy`, `forces`, `stress` when periodic) with the contract shapes/dtypes.
4. **Fine-tuning and DDP smokes.** (a) A multihead-replay fine-tuning run under `--engine v1` passes the P0-5 fine-tuning contract: it completes, the model contains the `pt_head` plus the new head, fine-tuning-set error decreases vs the unadapted model, and pseudolabel replay produces model-generated labels rather than file labels. (b) The TRN-4 (#1578) CPU-DDP smoke passes, and so does TRN-5 (#1639)'s 2-rank L-BFGS closure smoke.
5. **The must-have example.** An **arbitrary spherical-tensor observable declared purely in config** — spec fields: name, irreps shape, per-atom flag, units, normalization — is trained and evaluated end-to-end with **zero new code**: the spec triggers automatic head creation and an automatic loss term, and derivatives are auto-named (`d_<quantity>_d_pos`; `energy` special-cases to `forces`/`stress`). Declaring a property absent from the data must be a hard error. (MAG-1 (#1584) exercises this same mechanism on a real feature, including the third differentiation target `dE/dm`; the gate only needs a synthetic/toy observable.)
6. **The 68 black-box test functions pass under `--engine v1`.** The 68 functions (54 in `tests/workflows` + 14 in `tests/integrations`) re-run under `--engine v1` for the migrated capabilities and pass; non-migrated capabilities skip cleanly. They are routed through the single `tests/helpers.py` indirection to the launcher entry point (engine selected by env, `MACE_ENGINE=v1`) — the tests themselves are not rewritten. Capability-marked suites (`magnetic`, `les`, `polar`, `torchsim`, `schedulefree`, `network`, `bin_lammps` — the probes in `tests/conftest.py:60-69`) run under their own jobs with `MACE_REQUIRE_CAPS` set, so a guaranteed-but-broken capability fails rather than skips.
7. **Feature-inventory audit.** Every feature-inventory row (P0-0; tracked home `tests/golden/`) whose destination ticket lies in Phase 2–3 scope is audited: implementing tickets flipped their rows to `done`, and the gate verifies no row in scope is unresolved or silently dropped. Every deliberate omission must appear as an explicit DROP row with a reason — the Phase 3 DROP set is `--save_cpu`, `--force_mh_ft_lr`, `--plot_interaction_e`, the `--swa*` option-string spellings, the two `mace_*_cueq` console scripts, the pseudolabel batch-failure and function-level keep-going fallbacks, per-rank pseudolabel generation, and the `no_data_ok` unlabelled continuation. Rows exposing a gap become tickets before the gate closes.

**Task:**
1. Run criteria 1–6 on CPU and record the results (commands, seeds, error tables, diffs vs the committed references) in the gate PR description.
2. Perform the inventory audit (criterion 7): list the Phase 2–3 rows, verify their `done` status against merged tickets, confirm every DROP has a reason and a pinning test, and file follow-up tickets for any gap found.
3. For any failing criterion: file/route the fix to the owning ticket (TRN-*, CLI-1 (#1579), FT-*, ELEC-1 (#1591)/ELEC-2 (#1592), DEP-1 (#1583)/1a, MAG-1 (#1584), PAR-1 (#1572)), then re-run the gate. This ticket itself carries no feature fixes.
4. When all seven criteria hold, mark the phase gate passed and confirm the `git ls-files mace/` non-increase check is active in CI from this point on.

**Out of scope:** fixing any failing criterion (owned by the implementing tickets); flipping the default engine to `v1` (separate decision, parity-gated); the real-LAMMPS binary tier and LAMMPS export/checkpoint capability (DEP-2 (#1594)); foundation-model conversion, MACE-Polar's included (Phase 4, FM-2 (#1587)); the LES model and the split-charge / SCF families (ELEC-4 (#1635)/ELEC-3 (#1593), Phase 5 — only their calculator surfaces, where they exist at all, are in Phase 3 scope); regenerating any golden (goldens are never regenerated inside a feature or gate ticket).

**Acceptance criteria:**
- [ ] `mace train --engine v1` on the tiny task: loss decreases, checkpoint + resolved-config metadata written, resume works, final error table comparable to `tests/golden/references/tiny_scaleshift_training_errors.json`.
- [ ] PAR-1 (#1572)'s single-step gradient parity (fp64 `d(loss)/d(params)` allclose, byte-identical weights) is green.
- [ ] A two-stage Adam→L-BFGS run completes on the tiny task and its full-batch gradient matches the concatenated-dataset gradient at fp64.
- [ ] The legacy-vs-v1 comparison on the P0-5 end-to-end regression set (surfaces + molecules, dipoles + partial charges) is recorded, same metrics old-vs-new.
- [ ] `mace eval` and the new ASE calculator agree numerically per config on the contract keys.
- [ ] Multihead-replay fine-tuning smoke passes the P0-5 contract (completes; `pt_head` + new head present; error decreases; pseudolabels are model-generated), and both the CPU-DDP and the 2-rank L-BFGS smokes pass.
- [ ] A new spherical-tensor observable declared purely in config trains and evaluates end-to-end with zero new code, derivatives auto-named.
- [ ] The 68 black-box test functions pass under `--engine v1` for migrated capabilities; others skip cleanly, and each capability job runs with its `MACE_REQUIRE_CAPS` guarantee.
- [ ] The electrostatics core is green under `MACE_REQUIRE_CAPS=polar`, so a missing `[electrostatics]` extra fails the gate instead of skipping it: ELEC-1 (#1591)'s reference solver resolves once at build time with nothing dispatching in `forward`, its three-regime cell assertions hold, and ELEC-2 (#1592)'s ported `PolarMACE` reproduces the P0-3a polar golden (energy, forces, periodic stress, atomic dipole, polarizability) at the fp64 tolerance row.
- [ ] The 15 polar config flags (`mace/tools/arg_parser.py:311-395`) and the polar user surface — `mace polar density-cube`, the seven polar result keys, Fukui output, the polar Hessian branch — are reachable under `--engine v1`, and their inventory rows are `done` rather than deferred to Phase 5.
- [ ] All Phase 2–3 feature-inventory rows audited: `done`, an explicit DROP with a reason and a pinning test, or a filed follow-up ticket; none unresolved.
- [ ] Gate result recorded; the `git ls-files mace/` strictly-non-increasing CI check is active.

**Verify:**
```bash
MACE_ENGINE=v1 python -m pytest tests/workflows tests/integrations -n 2   # the 68 black-box test functions (84 collected items) under v1
python -m pytest tests/parity -m "not slow"                               # incl. single-step gradient parity
mace train --engine v1 --config <tiny-task-config>.yaml                   # then compare the final error table
#   against tests/golden/references/tiny_scaleshift_training_errors.json
MACE_REQUIRE_CAPS=polar python -m pytest tests/parity -m "not slow" -k polar   # ELEC-1 (#1591)/ELEC-2 (#1592) vs the P0-3a golden
python -m pytest tests/golden -k inventory                                # inventory checker (tracked home)
```
**Review focus:** the two judgment calls — whether the v1 error tables are genuinely "legacy-comparable" (backed by the gradient-parity green, not eyeballing alone) and whether the inventory audit is honest (no row quietly waved through, every DROP carrying a reason and a pinning test). Gate sign-off is not a solo merge.
